src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -532,18 +550,55 @@ protected List<Field> getFieldsForFloatVector(final float[] array, final FieldTy
      * Function returns a list of fields to be indexed when the vector is byte type.
      *
      * @param array array of bytes
-     * @param fieldType {@link FieldType}
      * @return {@link List} of {@link Field}
      */
-    protected List<Field> getFieldsForByteVector(final byte[] array, final FieldType fieldType) {
+    protected List<Field> getFieldsForByteVector(final byte[] array) {
         final List<Field> fields = new ArrayList<>();
-        fields.add(new VectorField(name(), array, fieldType));
+        fields.add(createVectorField(array, dimension));
         if (this.stored) {
             fields.add(createStoredFieldForByteVector(name(), array));
         }
         return fields;
     }
 
+    private Field createVectorField(float[] vectorValue, int dimension) {
+        final FieldType tempFieldType = new FieldType(fieldType);
```

Comment:
Why do we need to create this here as opposed to KNNFieldMapper Construction? This will create temp field per doc parsed right?

Reply:
This is a valid question. So, I have to do it because I have to add the vector in the fieldType and once the constructor a mapper is done it freeze the field. Here: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/MethodFieldMapper.java#L68, same is there for other mappers.

So, if we want to remove this temp field I have to remove the freeze part from all the mappers and then do it here. I think that is a better option and I can avoid the tempFieldType creation. 

Does that sound good?

Reply:
Im not sure Im following. We need to add the vector in the field not the field type. We can reuse the fieldType instead of recreating.

Reply:
@jmazanec15 yes we can reuse the fieldType for all the mappers except ModelFieldMapper. Because when the constructor of the ModelFieldMapper is called we just have modelId and as per this comment cluster state is not initialized at that time [ref](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/ModelFieldMapper.java#L49-L52) so we don't have access to information like VectorDataType, dimensions etc, which is required for Lucene Vector Field.


Now I see couple of ways here:
1. We can keep this code of creating the fieldType using a temp for every document ingestion for all the mappers.
2. We can skip doing the `fieldType.freeze()` for ModelFieldMapper and set attributes during [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/ModelFieldMapper.java#L64).
3. We use this tempFieldType behavior for only ModelFieldMapper.

Reply:
> We can skip doing the fieldType.freeze() for ModelFieldMapper and set attributes during [here](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/ModelFieldMapper.java#L64).

I would be in favior of this approach similar to how were handling the init's (i.e. initPerDimensionValidator)

Reply:
Let me take a look. Since the code has been updated now.

Reply:
Fixed.